### DISCUSSION TITLE
Dual stack for strongSwan Android client

### DISCUSSION THREAD

**Discussion by yuedongze** [strongswan/strongswan#481] [Q&A] [answered]
Hi there,

I’m trying to set up strongSwan app on my android phone as a road warrior that routes traffic through my laptop running the charon daemon.

I set up the server with an IPv4 address pool, local_ts set to 0.0.0.0/0 and the iptables rule. Everything works fine except that I notice some of the traffic on my phone browser uses IPv6 and they don’t get routes over the VPN link.

I tried to add a new address pool and a new traffic selector for the IPv6 use case (::/0) on the server side swanctl.conf, and now I see that the original web pages that used IPv6 did get routed through the link, but they were downgraded to IPv4. I also see the error “unable to install source route for %any6” from charon.

I’m wondering if dual stack is supported for the Android client? If so, is there any good example setup that I can refer to?

Config I had:

```
connections {

   rw-eap {
      pool = rw_pool, rw_v6_pool

      local {
         auth = pubkey
         certs = moonCert.pem
         id = moon.strongswan.org
      }
      remote {
         auth = eap-aka
         eap_id = %any
      }
      children {
         net {
            local_ts  = 0.0.0.0/0, ::/0
            esp_proposals = aes128gcm128-x25519
         }
      }
      version = 2
      send_certreq = no
      proposals = aes128-sha256-x25519
   }
}

pools {
  rw_pool {
    addrs = 10.3.0.0/24
  }
  rw_v6_pool {
    addrs = f3c3::/120
  }
}

secrets {

   eap-carol {
      id = carol
      secret = "Ar3etTnp01qlpOgb"
   }
}
```

Thanks!
Steven

**Comment by tobiasbrunner** [comment, marked as answer]
> I’m wondering if dual stack is supported for the Android client?

Sure. The problem is most likely on your server (the error happens there). Does it even have IPv6 connectivity? Or is your goal to NAT tunneled IPv6 traffic to IPv4? Also, you could just block IPv6 on the client while the IPv4-only VPN is established (see the advanced profile settings).

**Reply by yuedongze** [reply]
Hi Tobias, thanks for much for the quick response! I just realized that the server running in docker container on my Mac is IPv4 only. This explains why the v6 route failed. I decided to block all v6 traffic through the advanced settings and it now works great!

thanks again for the help.
